Framework/API/src/ExperimentInfo.cpp

```diff
@@ -256,10 +257,12 @@ makeDetectorInfo(const Instrument &oldInstr, const Instrument &newInstr) {
  * @param visitor : Component visitor to query. Visitor MUST have been filled
  * via registerContents.
  */
-void ExperimentInfo::makeAPIComponentInfo(const InfoComponentVisitor &visitor) {
+void ExperimentInfo::makeAPIComponentInfo(
+    const InfoComponentVisitor &visitor,
+    boost::shared_ptr<Beamline::DetectorInfo> detectorInfo) {
 
   // Internal Beamline ComponentInfo
-  m_componentInfo = visitor.componentInfo();
+  m_componentInfo = visitor.componentInfo(detectorInfo);
```

Comment:
I think as a follow up cleanup we can then move `DetectorInfo` into the visitor?